### PR TITLE
Minor doc update for DataOut with curved cells.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/numerics/data_out.h [hunk 1]
```diff
@@ -226,12 +253,17 @@ class DataOut : public DataOut_DoFData<DoFHandlerType, DoFHandlerType::dimension
    * using the mapping, i.e. a higher order mapping leads to a representation
    * of a curved boundary by using more subdivisions. Some mappings like
    * MappingQEulerian result in curved cells in the interior of the domain.
-   * However, there is nor easy way to get this information from the Mapping.
-   * Thus the last argument @p curved_region take one of three values
+   * The same is true if you have attached a manifold description to
+   * the cells of a triangulation (see @ref "Manifolds" manifold for more
+   * information). However, there is nor easy way to get query the mapping
```
Comment (tjhei) [inline include/deal.II/numerics/data_out.h]: "there is no easy way to query ... "

File: include/deal.II/numerics/data_out.h [hunk 2]
```diff
@@ -166,8 +166,35 @@ class DataOut : public DataOut_DoFData<DoFHandlerType, DoFHandlerType::dimension
...
+   * this can be emulated by plotting edges as a sequence of straight
+   * lines (and faces in 3d as a collection of bilinear patches) if
+   * DataOut::build_patches() is called with a number of subdivisions
+   * greater than 1.
+   *
+   * The elements of this enumeration then describe for which cells
+   * DataOut::build_patches() will query the manifold or boundary
+   * description for curved geometries:
+   *
+   * - @p no_curved_cells: The geometry or boundary description
```
Comment (tjhei) [inline include/deal.II/numerics/data_out.h]: Is there a reason you don't put this into the documentation of the individual members below?

**Comment by tjhei** [comment]
Looks good otherwise.

File: include/deal.II/numerics/data_out.h (hunk 2 above)
Comment (bangerth) [inline include/deal.II/numerics/data_out.h]: You mean the members of the enum? I guess I could do that.

Comment (tjhei) [inline include/deal.II/numerics/data_out.h]: Yes. Right now they are undocumented, so they won't even show up in the documentation.

**Pull request merged**